Adds Gorger, a new xenomorph

Walarks opened the pull request:
<!-- ***STOP!***  Read this: If this is not a PR ready for review and merge or WIP, open it as a draft PR, using the arrow next to 'Create Pull Request'>

<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

Adds new T3 Gorger

This xeno is supposed to achieve the niche of being a tank that can take marine fire for extended periods of time, while providing a lot of healing for other xenos and itself. To do so it needs to attack marines to gain and store blood, which forces it to expose itself to gain the full effect of its abilities.

Stats:
Health - 400 - 600
Overheal - 200 - 275
// Gorger's abilities grant overheal, which gets damaged before health
Resistances - 5 - 20%
Speed - a bit slower than young defiler, gets slowdown on weeds. 
// Off weeds should be pretty fast for a tanky xeno. Crusher without charging on weeds is slower than gorger on weeds.
Plasma - 200 - 400, regens up to 5% of max, can't be given plasma
// Plasma is referred to as blood. Abilities regenerate ~plasma~ blood
Slash Damage - 15 - 20
// 2 more than ancient drone's 18

Abilities:
 - Devour - eat a marine for transportation, can't eat revivable corpses
 - Psychic Link - channel and link to another xenomorph. When linked, you rest and take 50% of their damage. The link persists until you unrest, get too hurt or your partner moves too far away.
 - Drain - 15s CD - hold a marine in place and deal small damage over 2 seconds, gaining blood and healing up to 80. The marine can still shoot/do stuff during the duration and after the ability is broken it staggers them slightly. You also can't slash while performing this.
If cast on a non-revivable human, restores health, or overheal, at a faster rate than resting and sunder at normal rate, without granting blood. This uses the corpse's blood and you can only drain corpses which have sufficient blood.
- Transfusion - 2s CD - costs blood and restores 30% health or grants overheal for full health
 - Rejuvenate - 4s CD - drains blood continuosly, slows you down and reduces damage taken, while restoring health over time.
 - Carnage - 15s CD - you gain movement speed and heal up to 20% on your next attack, scaling with missing blood. If your blood is below a certain %, you also knockdown your victim and drain some blood, during which you can't move or attack.
 - Feast - 3m CD - drains blood each tick and restores 8% health, overheal at 50% efficiency when on full health, until no blood is left or cancelled early. Carnage also heals nearby xenos

## Why It's Good For The Game

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

I think this xeno's playstyle as a tank that heals in exchange of pure damage isn't currently found among other xenos. Defender is the closest that I can think of, but to me it seems to be more of a T1 warrior than a tank. Crusher is one of the most durable xenos by stats, but it also doesn't provide a reward for taking damage or being in the middle of a fight.

## Changelog
:cl:Slotbot
expansion: adds a new xeno caste
expansion: gives all xenos 150 max overheal to be used for gorger ability
fix: pre-spawned humans now have the non-defibbable trait
fix: attack animation no longer cancels other animations on same object as long as both have the ANIMATION_PARALLEL flag
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Walarks commented on code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:
```diff
@@ -0,0 +1,184 @@
+		to_chat(owner, "<span class='warning'>This won't do!</span>")
+		return FALSE
+	if(X.blood_bank >= 100)
+		to_chat(owner, "<span class='warning'>Somehow we feel sated for now...</span>")
```
applied

Walarks commented on code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:
```diff
@@ -0,0 +1,184 @@
+/datum/action/xeno_action/activable/rejuvenate
+	name = "Rejuvenate/Transfusion"
+	action_icon_state = "Rejuvenate"
+	mechanics_text = "When used on self drains 10 blood and restores 40% health/plasma over 5s. When used on another xenomorph - restores 30% of their health after 2s. When used on a dead human you grab them and heal by draining their blood"
```
applied

Walarks commented on code/datums/status_effects/buffs.dm:
```diff
@@ -0,0 +1,65 @@
+	X.adjustFireLoss(-damage)
+	if(X.has_status_effect(STATUS_EFFECT_XENO_FEAST))
+		for(var/mob/living/carbon/xenomorph/H in range(4, owner))
+			if(H != X)
```
updated

Walarks commented on code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:
```diff
@@ -0,0 +1,184 @@
+		return FALSE
+	return TRUE
+
+/datum/action/xeno_action/activable/drain/use_ability(mob/living/carbon/A)
```
changed

Walarks commented on code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:
```diff
@@ -0,0 +1,184 @@
+	A.drop_all_held_items()
+	X.visible_message(A, "<span class='danger'>The [X] grabs [A]!</span>")
+	A.SetImmobilized(2 SECONDS)
+	if(!do_after(X, 2 SECONDS, TRUE, A, BUSY_ICON_HOSTILE, ignore_turf_checks = FALSE))
```
I think it is better to do something even if you fail since it goes on cooldown anyways

Walarks commented on code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:
```diff
@@ -0,0 +1,184 @@
+	var/mob/living/carbon/xenomorph/X = owner
+	if(!.)
+		return
+	if(ishuman(target))
```
This is slower than rejuvenate. It is supposed to allow healing without needing to rest

Walarks commented on code/datums/status_effects/buffs.dm:
```diff
@@ -0,0 +1,65 @@
+/datum/status_effect/xeno_carnage/proc/carnage_slash(datum/source, mob/living/target, damage)
+	SIGNAL_HANDLER
+	var/mob/living/carbon/xenomorph/X = owner
+	X.blood_bank += min(5, 100 - X.blood_bank)
```
I think it fits the theme and don't see why gorger can't have something like this. ~also ravager got it after gorger got scrapped for no sprites~

Walarks commented on code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:
```diff
@@ -0,0 +1,184 @@
+			to_chat(X, "<span class='notice'>Beyond our reach...</span>")
+			return
+		target.adjustBruteLoss(-X.maxHealth*0.3)
+		target.adjustFireLoss(-X.maxHealth*0.3)
```
It heals a lot in exchange for low damage

Walarks commented on code/datums/status_effects/buffs.dm:
```diff
@@ -0,0 +1,65 @@
+	to_chat(owner, "<span class='notice'>We feel our wounds close up and plasma reserves refilling.</span>")
+	X.adjustBruteLoss(-X.maxHealth*0.1)
+	X.adjustFireLoss(-X.maxHealth*0.1)
+	X.gain_plasma(X.xeno_caste.plasma_max*0.25)
```
added CAN_BE_GIVEN_PLASMA check

theta006 commented:
> George.

the rouny for this new xeno

Walarks commented on code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:
```diff
@@ -0,0 +1,184 @@
+	else
+		to_chat(X, "<span class='notice'>We decide to end our feast early...</span>")
+		X.remove_status_effect(STATUS_EFFECT_XENO_FEAST)
+	succeed_activate()
```
removed option to remove status effect

AndroidMK2 commented:
I need this new xeno in game or I'll commit sad

hyper2snyper commented on code/datums/status_effects/buffs.dm:
```diff
@@ -0,0 +1,64 @@
+	to_chat(owner_xeno, span_notice("Blood bank: [owner_xeno.blood_bank]%"))
+	owner_xeno.adjustBruteLoss(-damage)
+	owner_xeno.adjustFireLoss(-damage)
+	if(owner_xeno.has_status_effect(STATUS_EFFECT_XENO_FEAST))
```
early return

Walarks replied:
changed

hyper2snyper commented on code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:
```diff
@@ -0,0 +1,188 @@
+
+/datum/action/xeno_action/activable/drain/can_use_ability(atom/A, silent = FALSE, override_flags)
+	. = ..()
+	var/mob/living/carbon/xenomorph/gorger/X = owner
```
put this right above the `if(X.blood_bank)` since you dont need it until then

Walarks replied:
changed

hyper2snyper commented on code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:
```diff
@@ -0,0 +1,188 @@
+			to_chat(X, span_notice("We need to be next to our meal."))
+			return FALSE
+		return TRUE
+	if(isxeno(target))
```
probably could flip this to make it nicer.

Walarks replied:
Do you mean placing the isxeno check above the ishuman check? I think the proc will get into the ishuman check more often, which is why I think it should be above.

TiviPlus replied:
you coul invert the checks (ie !isxeno) to early return earler

hyper2snyper commented on code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:
```diff
@@ -0,0 +1,188 @@
+	cooldown_timer = 40 SECONDS
+	plasma_cost = 100
+
+/datum/action/xeno_action/activable/carnage/can_use_ability(atom/A, silent = FALSE, override_flags)
```
does... this do anything?

Walarks replied:
removed plasma cost, the ability applies a buff and that's it

lKiseki commented:
sir wasn't this a halloween-themed xeno
this is not what i asked for

TiviPlus commented on code/modules/mob/living/carbon/xenomorph/castes/gorger/castedatum_gorger.dm:
```diff
@@ -0,0 +1,129 @@
+	carnage_plasma_gain = 4
+	feast_plasma_drain = 20
+
+	actions = list(
+		/datum/action/xeno_action/activable/drain,
+		/datum/action/xeno_action/activable/rejuvenate,
+		/datum/action/xeno_action/activable/carnage,
+		/datum/action/xeno_action/activable/feast,
```
missing rest?

Walarks replied:
added

TiviPlus commented on code/modules/admin/topic.dm:
```diff
@@ -426,6 +426,8 @@ Status: [status ? status : "Unknown"] | Damage: [health ? health : "None"]
 				newmob = M.change_mob_type(/mob/living/carbon/xenomorph/boiler, location, null, delmob)
 			if("crusher")
 				newmob = M.change_mob_type(/mob/living/carbon/xenomorph/crusher, location, null, delmob)
+			if("gorger")
+				newmob = M.change_mob_type(/mob/living/carbon/xenomorph/gorger, location, null, delmob)
```
doesnt this also need an admin ui update

Walarks replied:
removed for now

TiviPlus commented on code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:
```diff
@@ -0,0 +1,176 @@
+	if(!iscarbon(A))
+		return FALSE
+	if(!A.can_sting())
+		to_chat(owner, span_xenowarning("This won't do!"))
+		return FALSE
```
here doesnt check for silent and following too

also "this wont do" should be more descriptive

Walarks replied:
changed

TiviPlus commented on code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:
```diff
@@ -0,0 +1,176 @@
+		return TRUE
+	if(isxeno(target))
+		var/mob/living/carbon/xenomorph/gorger/X = owner
+		if(X == target)
+			if(!COOLDOWN_CHECK(X, rejuvenate_self_cooldown))
+				to_chat(X, span_notice("We need another [round(COOLDOWN_TIMELEFT(X, rejuvenate_self_cooldown) / 10)] seconds before we can revitalize ourselves."))
+				return FALSE
+			if(X.plasma_stored < self_plasma_cost)
+				to_chat(X, span_notice("We need [self_plasma_cost - X.plasma_stored]u more blood to revitalize ourselves."))
+				return FALSE
+		else
+			if(!X.line_of_sight(target) || get_dist(X, target) > 2)
+				to_chat(X, span_notice("It is beyond our reach, we must be close and our way clear."))
+				return FALSE
+			if(X.plasma_stored < ally_plasma_cost)
+				to_chat(X, span_notice("We need [ally_plasma_cost - X.plasma_stored]u more blood to restore a sister."))
+				return FALSE
+			if(isdead(target))
+				to_chat(X, span_notice("We can only help living sisters."))
+				return FALSE
+			if(!do_mob(X, target, 1 SECONDS, BUSY_ICON_FRIENDLY, BUSY_ICON_MEDICAL))
+				return FALSE
+		return TRUE
```
inverte this if/else and you can ealry return

Walarks replied:
changed

TiviPlus commented on code/modules/mob/living/carbon/xenomorph/castes/gorger/castedatum_gorger.dm:
```diff
@@ -0,0 +1,129 @@
+	drain_plasma_gain = 30
+	carnage_plasma_gain = 8
+
+/datum/xeno_caste/gorger/ancient
+	upgrade_name = "Ancient"
+	caste_desc = "Being within mere eyeshot of this hulking, dripping monstrosity fills you with a deep, unshakeable sense of unease."
```
where primordial :hampter:

TiviPlus commented on code/modules/mob/living/carbon/xenomorph/castes/gorger/castedatum_gorger.dm:
```diff
@@ -0,0 +1,129 @@
+
+/datum/xeno_caste/gorger/young
+	upgrade_name = "Young"
+
+	upgrade = XENO_UPGRADE_ZERO
```
```suggestion
	upgrade = XENO_UPGRADE_ZERO
```

TiviPlus commented on code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:
```diff
@@ -0,0 +1,176 @@
+	if(X.plasma_stored >= X.xeno_caste.plasma_max)
+		to_chat(X, span_xenowarning("No need, we feel sated for now..."))
+		return FALSE
+	if(get_dist(X, A) > 1)
+		to_chat(X, span_xenowarning("It is outside of our reach! We need to be closer!"))
+		return FALSE
```
why not use .Adjacent here?

Walarks replied:
changed

TiviPlus commented on code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:
```diff
@@ -0,0 +1,176 @@
+	if(get_dist(X, A) > 1)
+		to_chat(X, span_xenowarning("It is outside of our reach! We need to be closer!"))
+		return FALSE
+	return TRUE
+
```
```suggestion

```
. is implicitly true here

TiviPlus commented on code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:
```diff
@@ -0,0 +1,176 @@
+		if(H.stat != DEAD)
+			to_chat(owner, span_notice("We can only use still, dead hosts to heal."))
+			return FALSE
+		else if(get_dist(owner, H) > 1)
+			to_chat(owner, span_notice("We need to be next to our meal."))
+			return FALSE
```
ise adgacent here

 and silent checks throughout here

Walarks replied:
changed

TiviPlus commented on code/modules/mob/living/carbon/xenomorph/xeno_defines.dm:
```diff
@@ -342,6 +350,10 @@
 	var/fortify = 0
 	var/crest_defense = 0
 
+	// Gorger vars
+	///Cooldown for using rejuvenate on self
+	COOLDOWN_DECLARE(rejuvenate_self_cooldown)
```
make this a timer cooldown?

Walarks replied:
How would it benefit from being a timer cooldown?

Walarks replied:
changed to timer

TiviPlus commented on code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:
```diff
@@ -0,0 +1,176 @@
+
+/datum/action/xeno_action/activable/feast/use_ability()
+	var/mob/living/carbon/xenomorph/X = owner
+	if(X.has_status_effect(STATUS_EFFECT_XENO_FEAST))
+		to_chat(X, span_notice("We decide to end our feast early..."))
+		X.remove_status_effect(STATUS_EFFECT_XENO_FEAST)
+		return
```
cooldown is 180 seconds so you can only cancel 20 seconds early

and why would you want to cancel early anyway?

Walarks replied:
the buff drains ~plasma~ stored blood over time and you can cancel to not waste it

BraveMole replied:
what tivi said is still an issue

TiviPlus commented on code/datums/status_effects/buffs.dm:
```diff
@@ -0,0 +1,84 @@
+/datum/status_effect/xeno_rejuvenate
+	id = "xeno_rejuvenate"
+
+/datum/status_effect/xeno_rejuvenate/on_creation(mob/living/new_owner, set_duration)
```
I think these abiltities all use the default icon pls fix

Walarks replied:
fixed

Walarks replied:
added icon

TiviPlus commented on code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:
```diff
@@ -0,0 +1,176 @@
+				return FALSE
+		else
+			if(!X.line_of_sight(target) || get_dist(X, target) > 2)
+				to_chat(X, span_notice("It is beyond our reach, we must be close and our way clear."))
+				return FALSE
```
```suggestion
				to_chat(X, span_notice("It is beyond our reach, we must be close and our way must be clear."))
				return FALSE
```

TiviPlus commented on code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:
```diff
@@ -0,0 +1,176 @@
+// ***************************************
+/datum/action/xeno_action/activable/rejuvenate
+	name = "Rejuvenate/Transfusion"
+	action_icon_state = "Rejuvenate"
+	mechanics_text = "When used on self, drains blood and restores health over time. When used on another xenomorph, costs blood and restores some of their health. When used on a dead human, you heal gradually."
```
all these missiong action icons rn see actions.dmi we have a lot of unused ones you might find something

Walarks replied:
fixed

neohenry commented:
> 
> 
> sir wasn't this a halloween-themed xeno this is not what i was expecting

it's a winter-themed xeno because elsa paid for this, and she is winter

BraveMole commented on code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:
```diff
@@ -0,0 +1,253 @@
+		return
+	if(ishuman(target))
+		var/mob/living/carbon/human/H = target
+		if(H.stat != DEAD)
```
no synth check, no robo check. Probably should just check for the NO_BLOOD flag

TiviPlus replied:
There should be none, maybe synth but robo no

BraveMole commented on code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:
```diff
@@ -0,0 +1,253 @@
+	var/mob/living/carbon/xenomorph/X = owner
+	if(ishuman(A))
+		var/mob/living/carbon/human/target = A
+		while(X.health < X.maxHealth && do_after(X, 2 SECONDS, TRUE, A, BUSY_ICON_HOSTILE))
```
No max blood check

Walarks replied:
added check for blood and for defibbable

BraveMole reviewed (changes requested):
Not a fan of the ability to drain blood from defibabble humans.

BraveMole commented on code/datums/status_effects/xeno_buffs.dm:
```diff
@@ -23,3 +23,106 @@
+/datum/status_effect/xeno_carnage
+	id = "xeno_carnage"
+	alert_type = /obj/screen/alert/status_effect/xeno_carnage
+	///PLasma gain for each attack
```
```suggestion
	///Plasma gain for each attack
```

BraveMole commented on code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:
```diff
@@ -0,0 +1,258 @@
+		if(!silent)
+			to_chat(X, span_xenowarning("It is outside of our reach! We need to be closer!"))
+		return FALSE
+/datum/action/xeno_action/activable/drain/use_ability(mob/living/carbon/A)
```
```suggestion

/datum/action/xeno_action/activable/drain/use_ability(mob/living/carbon/A)
```

BraveMole reviewed (changes requested):
No keybind, no ai_should_use for gorger abilities

Ragecrabbo commented:
Some thoughts on gorgers after having played them a little.
1.Rejuvenate/Transfusion heal is incredibly powerful for healing other xenos for its cost, like shrike heal levels of good. For a combat oriented xeno that already has really good amount of self heals and a cooldown to heal other xenos from slashing this might be a bit too much.
2. Carnage and feast are functionally not very different from eachother apart from one costing blood while other gives you more and you will almost always want to use them together when you can. Might want to combine them into one ability or make some minor changes into one of them to make them more situational.
3. Gorger is very quiet. More sounds from using their abilities would be nice as sound plays a huge part in both xeno and marine gameplay and its often difficult to tell what other gorgers are doing without having a glance at the combat log.
4. Despite all the heals gorger has no real way to get rid of all the sunder it will recieve from marines shooting at it (honestly might be for the best but still very annoying for a tank xeno)

Walarks commented:
> Some thoughts on gorgers after having played them a little. 1.Rejuvenate/Transfusion heal is incredibly powerful for healing other xenos for its cost, like shrike heal levels of good. For a combat oriented xeno that already has really good amount of self heals and a cooldown to heal other xenos from slashing this might be a bit too much. 2. Carnage and feast are functionally not very different from eachother apart from one costing blood while other gives you more and you will almost always want to use them together when you can. Might want to combine them into one ability or make some minor changes into one of them to make them more situational. 3. Gorger is very quiet. More sounds from using their abilities would be nice as sound plays a huge part in both xeno and marine gameplay and its often difficult to tell what other gorgers are doing without having a glance at the combat log. 4. Despite all the heals gorger has no real way to get rid of all the sunder it will recieve from marines shooting at it (honestly might be for the best but still very annoying for a tank xeno)

1. I'll look into the healing. Currently shrike also removes a lot of debuffs and the heal scales with heal pheros(gorger's doesn't), but I'm not sure how much this affects things.
2. The main difference is that Feast has a big cooldown and heals without attacking. The issue with combining them is that I would need to sacrifice the heal over time or the heal on attacks. Or combine there and make it not give blood, but I think the current iteration is better. I think this is mostly an issue with carnage, which I could rework to have less duration and more impact.
3. Yeah, I'll add on-mob effects for carnage and Feast, and maybe some more sounds
4. Sunder is on purpose. It has very high health so it has less of a negative from sunder, but also no easy way to recover on its own.

DeltaFire15 commented:
It seems kinda odd for a tank, especially a T3 at that, to have zero bomb armor, given that quite a few explosives gib anything less than XENO_BOMB_RESIST_1 regardless of health values.

DeltaFire15 commented on code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:
```diff
@@ -0,0 +1,258 @@
+	var/mob/living/carbon/xenomorph/X = owner
+	if(A.stat == DEAD)
+		var/mob/living/carbon/human/target = A
+		while(X.health < X.maxHealth && do_after(X, 2 SECONDS, TRUE, target, BUSY_ICON_HOSTILE))
```
Since it's a hybrid health / sunder heal cycle, my suggestion would be allowing them drain a corpse to heal their sunder even if full on hp, since it's one of the two effects of this. Xenomorphs who for some reason don't want this behavior could always just cancel the drain once healed.
(was kinda weird to not have during the initial Gorger rounds)
```suggestion
		while((X.health < X.maxHealth || X.sunder) && do_after(X, 2 SECONDS, TRUE, target, BUSY_ICON_HOSTILE))
```

Walarks replied:
Here the sunder heal is similar to resting, while the health heal is much more than resting. Due to this, it is preferable to not drain bodies for sunder heal since this uses up their blood, while you can be resting for free instead if you wanted sunder recovery.

DeltaFire15 replied:
Prior to the most recent commits I'd have called it a valid tool for sunder recovery since it was still higher than base resting or retreating from non-weed areas to rest, though this is now effectively nulled by the number change. Feel free to just mark this resolved.

BraveMole commented on code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:
```diff
@@ -1,56 +1,165 @@
-		succeed_activate()
-		add_cooldown()
+
+/datum/action/xeno_action/activable/drain/use_ability(mob/living/carbon/target)
```
cast to human here

Walarks replied:
changed

BraveMole commented on code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:
```diff
@@ -1,56 +1,165 @@
 	succeed_activate()
 	add_cooldown()
 
+/datum/action/xeno_action/activable/drain/ai_should_use(atom/target)
```
you just need to return can_use_ability in that proc, nothing else.

Walarks replied:
changed

BraveMole commented on code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:
```diff
@@ -59,78 +168,69 @@
-		to_chat(X, span_notice("We tap into our reserves for nourishment."))
+/datum/action/xeno_action/activable/rejuvenate/use_ability(atom/target)
+	var/mob/living/carbon/xenomorph/owner_xeno = owner
+	if(target == owner_xeno)
```
early return, and since the plasma cost is zero on the ability itself you don't need succeed_activate()

Walarks replied:
changed

BraveMole commented on code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:
```diff
@@ -59,78 +168,69 @@
 
+/datum/action/xeno_action/activable/rejuvenate/ai_should_use(atom/target)
+	// no healing non-xeno
+	if(!isxeno(target))
```
This should be a self heal as well ? Since it's never targeting itself

Walarks replied:
What do I need to change for it to also heal itself?

BraveMole replied:
if(owner_health && can_use_ability)
    use_ability(owner)

something like that

Walarks replied:
added

BraveMole commented on code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:
```diff
@@ -140,14 +240,31 @@
+	var/mob/living/carbon/xenomorph/owner_xeno = owner
+	owner_xeno.apply_status_effect(/datum/status_effect/xeno_carnage, 20 SECONDS, owner_xeno.xeno_caste.carnage_plasma_gain)
 	add_cooldown()
+	succeed_activate()
```
no cost so no need

Walarks replied:
removed

BraveMole commented on code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:
```diff
@@ -140,14 +240,31 @@
+	var/mob/living/carbon/xenomorph/owner_xeno = owner
+	if(owner_xeno.plasma_stored > owner_xeno.xeno_caste.plasma_max * 0.8)
+		// ability is primarily used for plasma gain, so should be used even at full health
+		if(owner_xeno.health > owner_xeno.maxHealth * 0.9)
```
? you are returning false either way

Walarks replied:
changed check

BraveMole commented on code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:
```diff
@@ -140,14 +240,31 @@
+			return FALSE
+		return FALSE
+	// nothing gained by slashing allies
+	if(target.get_xeno_hivenumber() == owner_xeno.get_xeno_hivenumber())
```
you never slash allies.

Check if the target is a carbon though, cause the ai targets ai_nodes

Walarks replied:
added check

BraveMole commented on code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:
```diff
@@ -140,14 +240,31 @@
+	if(target.get_xeno_hivenumber() == owner_xeno.get_xeno_hivenumber())
+		return FALSE
+
+	return can_use_ability(target) ? TRUE : ..()
```
```suggestion
	return can_use_ability(target)
```

..() just return FALSE all the time. I know they are still some exemple that do that, but it's just stupid and it's code i still didn't remove

Walarks replied:
changed

BraveMole commented on code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:
```diff
@@ -156,103 +273,49 @@
 	COOLDOWN_START(src, misclick_prevention, 2 SECONDS)
-	succeed_activate()
 	add_cooldown()
+	succeed_activate()
```
ditto

Walarks replied:
removed

BraveMole commented on code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:
```diff
@@ -156,103 +273,49 @@
-	if(!HAS_TRAIT(victim, TRAIT_UNDEFIBBABLE))
-		if(!silent)
-			to_chat(owner, span_warning("This creature is struggling too much for us to devour it."))
+/datum/action/xeno_action/activable/feast/ai_should_use(atom/target)
```
check if target is atom AND not a friendly xeno

Walarks replied:
This ability doesn't care about target, why should I check it?

BraveMole commented on code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:
```diff
@@ -140,14 +240,31 @@
 	add_cooldown()
+	succeed_activate()
+
+/datum/action/xeno_action/activable/carnage/ai_should_use(atom/target)
```
check if is carbon

Walarks replied:
changed

BraveMole commented on code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:
```diff
@@ -156,103 +273,49 @@
-	victim.forceMove(X)
-	X.eaten_mob = victim
-	add_cooldown()
+	return can_use_ability(target) ? TRUE : ..()
```
ditto

Walarks replied:
changed

Walarks commented on code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:
```diff
@@ -1,56 +1,165 @@
+			return FALSE
+		return TRUE
+
+	if(!.)
```
This is here in order to be able to drain dead bodies while the ability is on cooldown.

BraveMole reviewed (changes requested):
OP, design a nerf

BraveMole reviewed (changes requested):
What did you change? There was no bug, just balance issues

Walarks commented:
> What did you change? There was no bug, just balance issues

Overall reduced max health by 200 and fixed all healing effects benefitting from double healing when target has both burn and brute, which should reduce healing across the board by a noticeable amount and a big amount when having both burn and brute. Also reduces available health at the start of an engage unless you use resources to get overheal prematurely.

Drain - Instead of instant, the benefits are gained over 4 seconds. Total damage down from 20 to 8, added 80 healing to it, holds the target, but doesn't knock down or cause to drop gun and you can't slash. When alone, even against a single target who can wield a gun and shoot you, it won't be very useful.

Rejuvenate/Transfusion - healing self was nerfed to 40% total health

Carnage - reworked. In cases where you are left alone to slash a lot, this could give a lot of free healing/blood, while not benefitting much if you can't reach anyone. Now it should be more slightly below middle.

Feast - added requirement to have at least 200 blood in order to use it and reduced healing per tick from 10% to 8%. The main issue last time was the blood requirement. I think having the blood requirement would make it harder to use immediately after it's off cooldown.

DeltaFire15 commented:
Gorgers appear to be unable to heal if they have overheal but aren't full health. This applies to all sources of healing and requires losing all overheal to fix.

neohenry commented:
Multiple gorgers increase the time for drain, meaning that if three gorgers are draining the same person, the person is snared for 12 seconds. Either give the ability for person to not be snared for that long or change ability to not lock person in place. Marines are salting about it.

DeltaFire15 commented:
> Multiple gorgers increase the time for drain, meaning that if three gorgers are draining the same person, the person is snared for 12 seconds. Either give the ability for person to not be snared for that long or change ability to not lock person in place. Marines are salting about it.

Xenomorph abilities that stun or block relocation (fling / acid spray / lunge / pounce / sneak attack / whatever else) can be chained after the first ends. Removing this from gorger only, whose stun doesn't even actually stun you (you can just shoot it and get it to back off, or, have friends with you to make it leave) seems kinda questionable. Sure, there's more Gorgers right now because they're new so this happening is more pronounced, but the same could be done with multiple of all the other stun castes.
(Besides the lock can also be cancelled, be it via the xeno being dislocated or you being dislocated, something that does not apply to any other "stun" abilities

Walarks commented:
> Multiple gorgers increase the time for drain, meaning that if three gorgers are draining the same person, the person is snared for 12 seconds. Either give the ability for person to not be snared for that long or change ability to not lock person in place. Marines are salting about it.

I could add a mechanic that doesn't allow the same person to be drained more than once every x amount of seconds, but I'll need to this about it. There are some other nerfs/reworks I have in mind in the meantime.

Lumipharon commented:
"12 second root" is a meme. Any T3 is going to rape you if there are three of them focusing you.
That said, losing movement for 4 seconds is very strong in this game. If you're in a small fight or marines are falling back, that 4 seconds is highly likely to mean death.

Walarks commented:
Changes since last testmerge:
- Reduced drain instances to 2 for 2 seconds of immobilize and adjusted overheal and plasma to match
- Removed regenerating up to 5% plasma passively
- Fixed rejuvenate casting multiple times

private-tristan commented:
It'd be nice if, upon using rejuvenate on another xeno, they got an effect similar to health infusion that gave them overheal, currently barely any gorgers ever use rejuvenate on others.

Walarks commented:
Changes since last testmerge:
> Removes gorger psydrain cost
> Reduces devour delays
> Fixes eating multiple humans
> Makes drain heal instead of granting overheal when damaged
> Makes draining dead humans grant overheal when on full health
> Makes carnage not execute when attacking non-humans

Walarks commented:
> It'd be nice if, upon using rejuvenate on another xeno, they got an effect similar to health infusion that gave them overheal, currently barely any gorgers ever use rejuvenate on others.

Yeah, I could do that later.

Walarks commented:
Actually, I'll do it right now.

For my reasoning this I thought this was a good idea before I get flamed for unkillable xenos, I think this can open up the ability to be used to give a xeno some extra shielding before an engage and offer a tactical choice between the gorger using blood for itself or another beno. This can also help with sieging as you could potentially give a queen more health to melt a cade while being shot.

novaepee commented:
when a xeno has

overheal
hivelord's beamu
resin for fire
shield pheno

BraveMole commented on code/__DEFINES/xeno.dm:
```diff
@@ -72,6 +72,12 @@ GLOBAL_LIST_INIT(xeno_ai_spawnable, list(
+#define XENO_IS_DAMAGED(xeno) (xeno.health < xeno.maxHealth)
+
+///Checks if xeno has damage or missing overheal
+#define XENO_IS_DAMAGED_FULL(xeno) (XENO_IS_DAMAGED(xeno) || xeno.overheal < xeno.xeno_caste.overheal_max)
```
these defines are not needed, they just make the code cryptic

Walarks replied:
removed

Walarks commented:
Changes since last update:
Using drain on a dead human is not affected by timeout triggered by taking damage
Reworked rejuvenate - rejuvenate now heals much less and instead reduces damage taken if the damage you are receiving each 2 seconds exceeds a certain amount, which should make gorger tankier against burst damage, while sustained damage should be less affected. The increased damage resistance is at the expense of less healing than before and added slowdown.

DeltaFire15 commented:
What exactly is the use case for an ability making a gorger *that* slow (besides worse Defender fortify, especially given Feast exists)
Note that this is taken off-weed, which is where Gorgers are faster than on weed.

![slow_gorgy](https://user-images.githubusercontent.com/46569814/145691027-9eeb50d9-8b33-412b-a3e5-1a543339880e.gif)

Walarks commented:
> What exactly is the use case for an ability making a Gorger _that_ slow (besides worse Defender fortify / Crest Defense, that also drains your limited resources, especially given Feast exists) Note that this is taken off-weed, which is where Gorgers are faster than on weed.
> 
> ![slow_gorgy](https://user-images.githubusercontent.com/46569814/145691027-9eeb50d9-8b33-412b-a3e5-1a543339880e.gif)
> 
> Fast (nonrejuv) Gorgy for reference: ![fast_gorgy](https://user-images.githubusercontent.com/46569814/145691111-04ae8374-1a2a-4d18-acdb-42ba1234b0ba.gif)

Gorger is supposed to be a tank, but currently dies too fast in situations where at the same time a tank is needed, such as siege. In this case, I had to choose between tankiness vs mobility, since if I have both it will be unkillable and able to easily harass marines, and I choose tankiness. Numbers could use more tweaking, but the point of the skill is to be used when going in to be able to soak up damage if you expect to be taking a lot, instead of being used when running away.

Walarks commented on code/__DEFINES/mobs.dm:
```diff
@@ -580,8 +580,12 @@ GLOBAL_LIST_INIT(xenoupgradetiers, list(XENO_UPGRADE_BASETYPE, XENO_UPGRADE_INVA
+#define GORGER_REJUVENATE_SELF_SLOWDOWN 6
+#define GORGER_REJUVENATE_SELF_AMOUNT 0.05
+#define GORGER_REJUVENATE_SELF_THRESHOLD 0.20
+#define GORGER_REJUVENATE_SELF_DMG_REDUCTION(damage) min(damage * 0.7, 40)
```
removed

BraveMole commented on code/datums/status_effects/xeno_buffs.dm:
```diff
@@ -141,8 +142,8 @@
 	if(get_dist(owner, target_mob) > link_range)
-		to_chat(target_mob, span_xenowarning("You have moved too far away from [owner]."))
-		to_chat(owner, span_xenowarning("[target_mob] has moved too far away."))
+		to_chat(target_mob, span_xenowarning("Are too far away from [owner]."))
```
doesn't mean anything

Walarks replied:
fixed
